modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/ContentDashboardAdminPortletTest.java

```diff
@@ -367,6 +367,183 @@ public void testGetProps() throws Exception {
 		}
 	}
 
+	@Test
+	public void testGetPropsWithAssetCategoriesAndChildAssetCategoriesSortedByKey()
+		throws Exception {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_company.getCompanyId(), _company.getGroupId(),
+				_user.getUserId());
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.fetchGroupVocabulary(
+				serviceContext.getScopeGroupId(), "audience");
+
+		AssetCategory assetCategory1 = _assetCategoryLocalService.addCategory(
+			_user.getUserId(), _company.getGroupId(), "A1",
+			assetVocabulary.getVocabularyId(), serviceContext);
+
+		AssetCategory assetCategory2 = _assetCategoryLocalService.addCategory(
+			_user.getUserId(), _company.getGroupId(), "A2",
+			assetVocabulary.getVocabularyId(), serviceContext);
+
+		AssetVocabulary childAssetVocabulary =
+			_assetVocabularyLocalService.fetchGroupVocabulary(
+				serviceContext.getScopeGroupId(), "stage");
+
+		AssetCategory childAssetCategory1 =
+			_assetCategoryLocalService.addCategory(
+				_user.getUserId(), _company.getGroupId(), "S1",
+				childAssetVocabulary.getVocabularyId(), serviceContext);
+
+		AssetCategory childAssetCategory2 =
+			_assetCategoryLocalService.addCategory(
+				_user.getUserId(), _company.getGroupId(), "S2",
+				childAssetVocabulary.getVocabularyId(), serviceContext);
+
+		try {
+			JournalArticle journalArticle1 = JournalTestUtil.addArticle(
+				_user.getUserId(), _group.getGroupId(), 0);
+
+			_journalArticleLocalService.updateAsset(
+				_user.getUserId(), journalArticle1,
+				new long[] {
+					assetCategory1.getCategoryId(),
+					assetCategory2.getCategoryId(),
+					childAssetCategory1.getCategoryId(),
+					childAssetCategory2.getCategoryId()
+				},
+				new String[0], new long[0], null);
+
+			JournalArticle journalArticle2 = JournalTestUtil.addArticle(
+				_user.getUserId(), _group.getGroupId(), 0);
+
+			_journalArticleLocalService.updateAsset(
+				_user.getUserId(), journalArticle2,
+				new long[] {
+					assetCategory1.getCategoryId(),
+					childAssetCategory2.getCategoryId()
+				},
+				new String[0], new long[0], null);
+
+			JournalArticle journalArticle3 = JournalTestUtil.addArticle(
```

Comment:
I think that maybe you can create a journal with their categories, using the service context.

Reply:
I think so, i needed to find it

Reply:
done